src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/OffenderTransactionHistoryRepository.java

```diff
@@ -11,9 +11,9 @@
 
 public interface OffenderTransactionHistoryRepository extends CrudRepository<OffenderTransactionHistory, Pk> {
 
-    @Query("select h from OffenderTransactionHistory h where h.offender.nomsId = ?1 and h.accountType = ?2 and (h.entryDate between ?3 and ?4)")
-    List<OffenderTransactionHistory> findForGivenAccountType(String offenderNo, String accountType, LocalDate fromDate, LocalDate toDate);
-
-    @Query("select h from OffenderTransactionHistory h where h.offender.nomsId = ?1 and (h.entryDate between ?2 and ?3)")
-    List<OffenderTransactionHistory> findForAllAccountTypes(String offenderNo, LocalDate fromDat, LocalDate toDate);
+    @Query("select h from OffenderTransactionHistory h where h.offender.nomsId = :offenderNo and " +
+        " (:accountType is null or h.accountType = :accountType) and " +
```

Comment:
Move to a single repository method with optional parameters due to the filter combinations getting out of hand.